bundle/deploy/terraform/import.go

```diff
@@ -0,0 +1,129 @@
+package terraform
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/log"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type BindOptions struct {
+	AutoApprove  bool
+	ResourceType string
+	ResourceKey  string
+	ResourceId   string
+}
+
+type importResource struct {
+	opts *BindOptions
+}
+
+// Apply implements bundle.Mutator.
+func (m *importResource) Apply(ctx context.Context, b *bundle.Bundle) error {
+	tf := b.Terraform
+	if tf == nil {
+		return fmt.Errorf("terraform not initialized")
+	}
+
+	err := tf.Init(ctx, tfexec.Upgrade(true))
+	if err != nil {
+		return fmt.Errorf("terraform init: %w", err)
+	}
+
+	importsFilePath, err := m.writeImportsFile(ctx, b)
+	if err != nil {
+		return fmt.Errorf("write imports file: %w", err)
+	}
+	log.Debugf(ctx, "imports.tf file written to %s", importsFilePath)
+
+	buf := bytes.NewBuffer(nil)
+	tf.SetStdout(buf)
+	changed, err := tf.Plan(ctx)
+	if err != nil {
+		return fmt.Errorf("terraform plan: %w", err)
+	}
+
+	if changed && !m.opts.AutoApprove {
+		cmdio.LogString(ctx, buf.String())
+		ans, err := cmdio.AskYesOrNo(ctx, "Confirm import changes? Changes will be remotely only after running 'bundle deploy'.")
+		if err != nil {
+			return err
+		}
+		if !ans {
+			// remove imports.tf file
+			_ = os.Remove(importsFilePath)
+			return fmt.Errorf("import aborted")
+		}
+	}
+
+	log.Debugf(ctx, "resource imports approved")
+	return nil
+}
+
+func (m *importResource) writeImportsFile(ctx context.Context, b *bundle.Bundle) (string, error) {
+	// Write imports.tf file to the terraform root directory
+	dir, err := Dir(ctx, b)
+	if err != nil {
+		return "", err
+	}
+
+	importsFilePath := filepath.Join(dir, "imports.tf")
+	f, err := os.Create(importsFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	_, err = f.WriteString(fmt.Sprintf(`import {
+	to = %s.%s
+	id = "%s"
+}`, m.opts.ResourceType, m.opts.ResourceKey, m.opts.ResourceId))
```

Comment:
Can we use the JSON representation instead? We use it for the intended state as well.